homeassistant/components/unifi/sensor.py

```diff
@@ -192,6 +234,41 @@ def async_device_state_value_fn(hub: UnifiHub, device: Device) -> str:
     return DEVICE_STATES[device.state]
 
 
+def make_device_temperatur_sensors() -> tuple[UnifiSensorEntityDescription, ...]:
+    """Create device temperature sensors."""
+
+    print("TÆÆÆÆST")
+
+    def make_device_temperature_entity_description(
+        name: str,
+    ) -> UnifiSensorEntityDescription:
+        return UnifiSensorEntityDescription[Devices, Device](
+            key=f"Device {name} temperature",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            entity_registry_enabled_default=False,
+            api_handler_fn=lambda api: api.devices,
+            available_fn=async_device_available_fn,
+            device_info_fn=async_device_device_info_fn,
+            name_fn=lambda device: f"{device.name} {name} Temperature",
+            object_fn=lambda api, obj_id: api.devices[obj_id],
+            supported_fn=partial(async_device_temperatures_supported_fn, name),
+            unique_id_fn=lambda hub, obj_id: f"{name.lower}_temperature-{obj_id}",
+            value_fn=partial(async_device_temperatures_value_fn, name),
+        )
+
+    return tuple(
+        make_device_temperature_entity_description(name)
+        for name in (
+            "CPU",
+            "Local",
+            "PHY",
+        )
```

Comment:
We have the same thing here as in https://github.com/home-assistant/core/pull/116737

We have a list of possible values. 

Would be nice if they was generated on the fly. 
But I'm not sure how we can do it, as the device need to provide data before we can make them.

Reply:
How do you mean generated on the fly? In all cases we wait for data before creating entities

Reply:
That is what I mean. 
It would be nice not to define those cases like `CPU`, `Local` and `PHY` here. 

I think we also briefly talked about it in #116737 that we could make those entities based on the data coming in. 
But I'm not sure where and how to start.

Reply:
If you know the structure you could iterate on the content of them and use the keys as name I guess? If I understand you correctly